rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -1242,3 +1423,263 @@ func addrFromSignerConfig(config *config.SignerConfig) (common.Address, error) {
 		return common.Address{}, fmt.Errorf("failed to determine signer address, unknown signer type: %v", config.SignerType)
 	}
 }
+
+// sanityChecksBeforeConstructingTransaction performs sanity checks before constructing a transaction.
+func (r *Layer2Relayer) sanityChecksBeforeConstructingTransaction(batchesToSubmit []*dbBatchWithChunks) error {
```

Comment:
I suggest splitting the sanity checks to a independent fie: `12_reayer_santiy.go`, currently this file `l2_relayer.go` is too long